apidef/oas/authentication.go

```diff
@@ -13,6 +13,16 @@ import (
 	"github.com/TykTechnologies/tyk/apidef"
 )
 
+// ValidateSecurityProcessingMode validates the security processing mode value.
+func ValidateSecurityProcessingMode(mode string) bool {
+	return mode == "" || mode == "legacy" || mode == "compliant"
```

Comment:
This line shouts for constants like:
```go
SecurityProcessingModeNone string = ""
SecurityProcessingModeLegacy string = "legacy"
SecurityProcessingModeCompliant string = "compliant"
```

Those constants can be used in the entire PR